### PR TITLE
Don't create duplicate notifications when UserSignup status update fails.

### PR DESCRIPTION
**Pull request by sbryzak**
This PR prevents duplicate deactivating and deactivated notifications from being created if they have already been previously created in cases where the UserSignup status failed to update.

Fixes https://issues.redhat.com/browse/CRT-1243

### CODE REVIEW

File: controllers/usersignup/usersignup_controller_test.go [hunk 1]
```diff
@@ -1839,6 +1839,78 @@ func TestUserSignupDeactivatedAfterMURCreated(t *testing.T) {
 		require.Equal(t, "https://registration.crt-placeholder.com", notification.Spec.Context["RegistrationURL"])
 		assert.Equal(t, "userdeactivated", notification.Spec.Template)
 	})
+
+	t.Run("second deactivated notification not created when it already exists", func(t *testing.T) {
```
Comment (alexeykazakov) [inline controllers/usersignup/usersignup_controller_test.go:1843]: What about a similar test for the second deactivating notification case?

**Review by alexeykazakov — COMMENTED** [review]
Looks good. I just want to make sure we are not missing tests here. See my comment.

File: controllers/usersignup/usersignup_controller_test.go (hunk 1 above)
Comment (sbryzak) [reply controllers/usersignup/usersignup_controller_test.go:1843]: Added

**Pull request merged**